runtime/compiler/env/VMJ9Server.cpp

```diff
@@ -1417,3 +1417,113 @@ TR_J9ServerVM::dereferenceStaticFinalAddress(void *staticAddress, TR::DataType a
+         {
+         TR_ASSERT(false, "Failed to validate addStackWalkerMaySkipFramesRecord\n");
+         }
+      }
+   return skipFrames;
+   }
+
+bool
+TR_J9SharedCacheServerVM::isMethodEnterTracingEnabled(TR_OpaqueMethodBlock *method)
+   {
+   // We want to return the same answer as TR_J9VMBase unless we want to force it to allow tracing
+   return TR_J9VMBase::isMethodEnterTracingEnabled(method) || TR::Options::getAOTCmdLineOptions()->getOption(TR_EnableAOTMethodEnter);
+   }
+
+bool
+TR_J9SharedCacheServerVM::isMethodExitTracingEnabled(TR_OpaqueMethodBlock *method)
+   {
+   return TR_J9VMBase::isMethodExitTracingEnabled(method) || TR::Options::getAOTCmdLineOptions()->getOption(TR_EnableAOTMethodExit);
+   }
+
+bool
+TR_J9SharedCacheServerVM::traceableMethodsCanBeInlined()
+   {
+   return true;
+   }
+
+bool
+TR_J9SharedCacheServerVM::canMethodEnterEventBeHooked()
+   {
+   // We want to return the same answer as TR_J9ServerVM unless we want to force it to allow tracing
+   return TR_J9ServerVM::canMethodEnterEventBeHooked() || TR::Options::getAOTCmdLineOptions()->getOption(TR_EnableAOTMethodEnter);
+   }
+
+bool
+TR_J9SharedCacheServerVM::canMethodExitEventBeHooked()
+   {
+   // We want to return the same answer as TR_J9ServerVM unless we want to force it to allow tracing
+   return TR_J9ServerVM::canMethodExitEventBeHooked() || TR::Options::getAOTCmdLineOptions()->getOption(TR_EnableAOTMethodExit);
+   }
+
+bool
+TR_J9SharedCacheServerVM::methodsCanBeInlinedEvenIfEventHooksEnabled()
+   {
+   return true;
+   }
+
+int32_t
+TR_J9SharedCacheServerVM::getJavaLangClassHashCode(TR::Compilation * comp, TR_OpaqueClassBlock * clazzPointer, bool &hashCodeComputed)
+   {
+   hashCodeComputed = false;
+   return 0;
+   }
+
+bool
+TR_J9SharedCacheServerVM::javaLangClassGetModifiersImpl(TR_OpaqueClassBlock * clazzPointer, int32_t &result)
+   {
+   return false;
+   }
+
+uint32_t
+TR_J9SharedCacheServerVM::getInstanceFieldOffset(TR_OpaqueClassBlock * classPointer, char * fieldName, uint32_t fieldLen,
+                                    char * sig, uint32_t sigLen, UDATA options)
+   {
+   TR::Compilation* comp = TR::comp();
+   TR_ASSERT(comp, "Should be called only within a compilation");
+
+   bool validated = false;
+
+   if (comp->getOption(TR_UseSymbolValidationManager))
+      {
+      if (comp->getSymbolValidationManager()->verifySymbolHasBeenValidated(static_cast<void *>(classPointer)))
+         {
+         validated = true;
+         }
+      else
+         {
+         TR_ASSERT(false, "Class 0x%p should already be validated\n", classPointer);
+         }
+      }
+   else
+      {
+      validated = ((TR_ResolvedRelocatableJ9Method*) comp->getCurrentMethod())->validateArbitraryClass(comp, (J9Class*)classPointer);
+      }
+
+   if (validated)
+      return this->TR_J9ServerVM::getInstanceFieldOffset (classPointer, fieldName, fieldLen, sig, sigLen, options);
+
+   return ~0;
```

Comment:
Heh, if this is not a typo, maybe it should be fixed to just be `return -1`.

Reply:
The function returns an unsigned data type so ~0 makes more sense here.